app/gui/view/debug_scene/cloud-dashboard/cloud-view/src/id.rs

```diff
@@ -0,0 +1,302 @@
+}
+
+
+
+// ========================
+// === ProjectIdVariant ===
+// ========================
+
+/// Marker struct that indicates an identifier is for a [`Project`].
+///
+/// [`Project`]: crate::project::Project
+#[derive(Clone, Copy, Debug)]
+#[non_exhaustive]
+pub struct ProjectIdVariant(PhantomData<()>);
+
+
+// === Trait `impl`s ===
+
+impl IdVariant for ProjectIdVariant {
+    const PREFIX: &'static str = "proj";
+}
+
+
+
+// ==========
+// === Id ===
+// ==========
+
+/// An identifer that is used to uniquely identify entities in the Cloud backend (e.g.,
+/// [`Project`]s).
+/// 
+/// Internally, this is a [K-Sortable Unique IDentifier] (KSUID). KSUIDs are a variant of unique
+/// identifier that have some nice properties that are useful to us. For example, KSUIDs are
+/// naturally ordered by generation time because they incorporate a timestamp as part of the
+/// identifier. This is desirable because it means that entities are stored in our database sorted
+/// by their creation time, which means we don't have to sort our entities after fetching them.
+///
+/// Example:
+/// ```
+/// # use enso_prelude::*;
+/// let id = enso_cloud_view::id::ProjectId::from_str("proj-27xJM00p8jWoL2qByTo6tQfciWC").unwrap();
+/// ```
+/// 
+/// An [`Id`] string contains: the type of the entity this identifier is for (e.g.
+/// [`ProjectId::PREFIX`] for [`Project`], [`OrganizationId::PREFIX`] for organization, etc.),
+/// followed by the [`ID_PREFIX_SEPARATOR`], then the base62-encoded KSUID.
+///
+/// Using KSUIDs for identifiers gives several useful properties:
+/// 1. Naturally ordered by generation time;
+/// 2. Collision-free, coordination-free, dependency-free;
+/// 3. Highly portable representations.
+///
+/// [K-Sortable Unique IDentifier]: https://github.com/segmentio/ksuid#what-is-a-ksuid
+/// [`Project`]: crate::project::Project
+#[derive(Clone, Copy, Derivative, Display)]
+#[display(bound = "T: IdVariant")]
+#[display(fmt = "{}{}{}", "T::PREFIX", "ID_PREFIX_SEPARATOR", "ksuid")]
+#[derivative(PartialEq, PartialOrd, Eq, Ord)]
+pub struct Id<T> {
+    /// The inner [`svix_ksuid::Ksuid`] value for this identifier.
+    pub ksuid:  svix_ksuid::Ksuid,
+    /// Denotes the target entity type for this identifier; for internal use.
+    #[derivative(PartialEq = "ignore", PartialOrd = "ignore", Ord = "ignore")]
+    pub marker: PhantomData<T>,
+}
+
+
+// === Trait `impl`s` ===
+
+impl<T> serde::Serialize for Id<T>
+where Id<T>: Display
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where S: serde::Serializer {
+        serializer.serialize_newtype_struct(ID_STRUCT_NAME, &self.to_string())
+    }
+}
+
+impl<T> FromStr for Id<T>
+where T: IdVariant
+{
+    type Err = crate::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let ksuid_str = match s.split_twice(T::PREFIX, ID_PREFIX_SEPARATOR) {
+            Some(("", "", s)) => s,
+            _ => Err(format!("\"{s}\" is not an {ID_STRUCT_NAME}."))?,
```

Comment:
This really asks for `anyhow!`.

Reply:
Will change in follow-up PR with switch to anyhow instead of boxed error.